airflow/providers/amazon/aws/transfers/gcs_to_s3.py

```diff
@@ -147,6 +152,9 @@ def execute(self, context: 'Context') -> List[str]:
             aws_conn_id=self.dest_aws_conn_id, verify=self.dest_verify, extra_args=self.dest_s3_extra_args
         )
 
+        if not self.keep_directory_structure and self.prefix:
+            self.dest_s3_key = os.path.join(self.dest_s3_key, self.prefix)
```

Comment:
will it work properly on Windows where `os.path.join` uses backslashes (`\`)? I think it should use `posixpath` instead

Reply:
Feel free to make PR. Airflow only works on POSIX (for multiple reasons) but you are right it would be beter to even hardcode "/'".

Reply:
Will remember to use `posixpath` going forward. Actually, I referred to some of the code [here](https://github.com/apache/airflow/blob/main/airflow/providers/google/cloud/transfers/gcs_to_sftp.py#L171) when I did my PR..